src/Banner.js

```diff
@@ -1,51 +1,50 @@
 /** @jsx jsx */
 import { jsx } from '@emotion/core'
-import styled from '@emotion/styled'
-import F from 'futil'
-import _ from 'lodash/fp'
-import { defaultProps } from 'recompose'
 import { Icon, Text, Flex } from '.'
 import theme from './theme'
+import { getVariant } from './utils'
 let { colors } = theme
 
-let BaseBanner = ({ children, icon = 'priority_high', ...props }) => (
-  <Flex
-    alignItems="center"
-    justifyContent="center"
-    css={{ padding: theme.spaces.xs }}
-    {...props}
-  >
-    <Icon
-      icon={icon}
-      style={{ marginRight: theme.spaces.xs, verticalAlign: 'text-bottom' }}
-    />
-    <Text small>{children}</Text>
-  </Flex>
-)
-
-let convertBannerStyles = ({ backgroundColor, textColor, icon }) =>
-  styled(defaultProps({ icon })(BaseBanner))({
-    backgroundColor,
-    '&, span, a': { color: `${textColor} !important` },
-  })
+let styleFromVariant = ({ backgroundColor, textColor }) => ({
+  backgroundColor,
+  '&, span, a': { color: `${textColor} !important` },
+})
 
-let bannerStyles = _.mapValues(convertBannerStyles, {
-  Warning: {
+let variants = {
+  regular: {
+    backgroundColor: colors.primary,
+    textColor: colors.backgrounds[0],
+    icon: 'priority_high',
+  },
+  warning: {
     backgroundColor: colors.warning,
     textColor: colors.secondary,
     icon: 'warning',
   },
-  Error: {
+  error: {
     backgroundColor: colors.error,
     textColor: colors.backgrounds[0],
     icon: 'report',
   },
-})
+}
 
-let Banner = convertBannerStyles({
-  backgroundColor: colors.primary,
-  textColor: colors.backgrounds[0],
-})
+let Banner = ({ children, icon, ...props }) => (
+  <Flex
+    alignItems="center"
+    justifyContent="center"
+    css={[
+      { padding: theme.spaces.xs },
+      styleFromVariant(getVariant(props, variants, 'regular')),
+    ]}
+    gap="xs"
+    {...props}
+  >
+    <Icon
+      icon={icon || getVariant(props, variants, 'regular').icon}
```

Comment:
We should cache the variant in a local variable so we don’t run it twice if there’s an icon